specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -47,34 +50,47 @@ import { Tree } from './Tree';
 export function TreeViewWrapper(): JSX.Element | null {
   useMenuItem('trees');
   const { tableName = '' } = useParams();
-  const treeName = getTable(tableName)?.name;
+  const treeName = getTable(tableName)?.name as
+    | AnyTree['tableName']
+    | undefined;
+
   const [treeDefinitions] = usePromise(treeRanksPromise, true);
   useErrorContext('treeDefinitions', treeDefinitions);
 
-  const treeDefinition =
-    typeof treeDefinitions === 'object' &&
-    typeof treeName === 'string' &&
-    isTreeTable(treeName)
-      ? caseInsensitiveHash(treeDefinitions, treeName)
+  const treeDefinitionArray =
+    typeof treeDefinitions === 'object' && typeof treeName === 'string'
+      ? (caseInsensitiveHash(treeDefinitions, treeName) as RA<{
+          readonly definition: SerializedResource<GeographyTreeDef>;
+          readonly ranks: RA<SerializedResource<GeographyTreeDefItem>>;
+        }>)
       : undefined;
 
-  if (treeName === undefined || !isTreeTable(treeName)) return <NotFoundView />;
-  return (
+  const [
+    currentDefinitionName = treeDefinitionArray === undefined
+      ? undefined
```

Comment:
hmm, I'm not sure this is legal js syntax - assignment to undefined is not allowed